pymeasure/display/widgets.py

```diff
@@ -1036,3 +1039,218 @@ def browse_triggered(self):
         path = QtGui.QFileDialog.getExistingDirectory(self, 'Directory', '/')
         if path != '':
             self.setText(path)
+
+
+class EstimatorThread(StoppableQThread):
+    new_estimates = QtCore.QSignal(list)
+
+    def __init__(self, get_estimates_callable):
+        StoppableQThread.__init__(self)
+
+        self._get_estimates = get_estimates_callable
+
+        self.delay = 2
+
+    def __del__(self):
+        self.wait()
+
+    def run(self):
+        self._should_stop.clear()
+
+        while not self._should_stop.wait(self.delay):
+            estimates = self._get_estimates()
+            self.new_estimates.emit(estimates)
+
+
+class EstimatorWidget(QtGui.QWidget):
+    """
+    Widget that allows to display up-front estimates of the measurement
+    procedure.
+
+    This widget relies on a `get_estimates` method of the `Procedure` class.
+    `get_estimates` is expected to return a list of tuples, where each tuple
+    contains two strings: a label and the estimate.
+
+    If the `SequencerWidget` is also used, it is possible to ask for the
+    current sequencer or its length by asking for two keyword arguments in the
+    Implementation of the `get_estimates` function: `sequence` and
+    `sequence_length`, respectively.
+
+    """
+    provide_sequence = False
+    provide_sequence_length = False
+    number_of_estimates = 0
+    sequencer = None
+
+    def __init__(self, parent=None):
+        super().__init__(parent)
+        self._parent = parent
+        log.info("doing nothing")
+
+        self.check_get_estimates_signature()
+
+        self.update_thread = EstimatorThread(self.get_estimates)
+        self.update_thread.new_estimates.connect(self.display_estimates)
+
+        self._setup_ui()
+        self._layout()
+
+        self.update_estimates()
+
+        self.update_box.setCheckState(1)
+
```

Comment:
`PySide2` does not like this :(, could You please replace with: `self.update_box.setCheckState(QtCore.Qt.CheckState.Checked)`

Reply:
Fixed it; Also changed the comparison of the `self.update_box.checkState()` with integers to a comparison with these enumerators, doesn't seem to be required but is more readable.